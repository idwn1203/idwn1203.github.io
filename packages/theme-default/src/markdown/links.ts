import type { Plugin } from 'unified'
import { visit } from 'unist-util-visit'

// parse [foo](bar) to <Link label="foo" to="bar" />
const anchors: Plugin<any[], any> = () => {
  return tree => {
    visit(tree, (node, idx, parent) => {
      if (node.type === 'link') {
        parent.children.splice(idx, 1, {
          type: 'html',
          value: `<Link to="${node.url}" label="${node.children[0].value}" />`,
        })
      }
    })
  }
}

export default anchors
