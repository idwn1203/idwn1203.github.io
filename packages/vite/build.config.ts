import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig({
  rollup: {
    resolve: {},
    inlineDependencies: true,
  },
  failOnWarn: false,
})
