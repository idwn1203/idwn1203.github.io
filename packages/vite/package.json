{
  "name": "@sveltepress/vite",
  "homepage": "https://sveltepress.site/",
  "version": "0.33.2",
  "description": "Sveltepress plugin for vite",
  "files": [
    "dist",
    "types.d.ts"
  ],
  "type": "module",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": "./dist/index.mjs",
    "./types": {
      "types": "./types.d.ts"
    }
  },
  "scripts": {
    "test": "vitest",
    "build": "unbuild",
    "prepare": "pnpm build"
  },
  "keywords": [
    "vite",
    "vite plugin",
    "sveltepress",
    "sveltepress vite",
    "sveltekit"
  ],
  "author": {
    "name": "Dongsheng Zhao",
    "email": "1197160272@qq.com"
  },
  "repository": {
    "type": "github",
    "url": "https://github.com/Blackman99/sveltepress",
    "directory": "packages/vite"
  },
  "license": "MIT",
  "devDependencies": {
    "@sveltepress/theme-default": "workspace:*",
    "@sveltejs/kit": "^2.5.1",
    "@types/cross-spawn": "^6.0.6",
    "@types/estree": "^1.0.5",
    "@types/fs-extra": "^11.0.4",
    "@types/mdast": "^4.0.3",
    "@types/node": "^20.11.20",
    "dayjs": "^1.11.10",
    "svelte": "^4.2.12",
    "unbuild": "^2.0.0",
    "vite": "^5.1.4",
    "vitest": "^1.3.1"
  },
  "dependencies": {
    "cross-spawn": "^7.0.3",
    "fs-extra": "^11.2.0",
    "lru-cache": "^10.2.0",
    "picocolors": "^1.0.0",
    "rehype-stringify": "^10.0.0",
    "remark-directive": "^3.0.0",
    "remark-emoji": "^4.0.1",
    "remark-extract-frontmatter": "^3.2.0",
    "remark-frontmatter": "^5.0.0",
    "remark-gfm": "^4.0.0",
    "remark-parse": "^11.0.0",
    "remark-rehype": "^11.1.0",
    "remark-stringify": "^11.0.0",
    "shiki": "^1.1.7",
    "svelte-preprocess": "^5.1.3",
    "unified": "^11.0.4",
    "unist-util-visit": "^5.0.0",
    "vfile": "^6.0.1",
    "vite-plugin-inspect": "^0.8.3",
    "yaml": "^2.4.0"
  },
  "peerDependencies": {
    "@sveltejs/kit": "^1.20.4 || ^2.0.0",
    "@sveltejs/vite-plugin-svelte": "^3.0.1",
    "svelte": "^3.55.1 || ^4.0.0",
    "vite": "^4.3.9 || ^5.0.0"
  }
}
