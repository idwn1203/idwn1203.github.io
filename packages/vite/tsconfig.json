{
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "skipLibCheck": true,
    "strict": true,
    "declaration": true,
    "outDir": "dist",
    "moduleResolution": "node",
    "removeComments": false,
    "module": "ESNext",
  },
  "include": ["src", "types.d.ts"]
}