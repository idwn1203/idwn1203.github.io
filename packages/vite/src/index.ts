import type { PluginOption } from 'vite'
import { sveltekit } from '@sveltejs/kit/vite'
import vitePluginInspect from 'vite-plugin-inspect'
import SveltepressVitePlugin from './plugin.js'
import type { Highlighter, LoadTheme, ResolvedTheme, SiteConfig, SveltepressVitePluginOptions } from './types'
import mdToSvelte from './markdown/md-to-svelte'

export * as log from './utils/log.js'

const sveltepress: (options: SveltepressVitePluginOptions) => PluginOption = ({
  theme,
  addInspect,
  siteConfig,
  remarkPlugins,
  rehypePlugins,
} = {
  addInspect: false,
}) => {
  const requiredSiteConfig: Required<SiteConfig> = {
    title: siteConfig?.title || 'Untitled site',
    description: siteConfig?.description || 'Build by Sveltepress',
  }
  const corePlugin = [
    SveltepressVitePlugin({
      theme,
      siteConfig: requiredSiteConfig,
      remarkPlugins,
      rehypePlugins,
    }),
    sveltekit(),
  ]

  const plugins = typeof theme?.vitePlugins === 'function'
    ? theme.vitePlugins(corePlugin)
    : [
        theme?.vitePlugins,
        ...corePlugin,
      ]
  if (addInspect)
    plugins.unshift(vitePluginInspect())

  return plugins
}

export { sveltepress, mdToSvelte }
export type { LoadTheme, Highlighter, SiteConfig, SveltepressVitePluginOptions, ResolvedTheme }
