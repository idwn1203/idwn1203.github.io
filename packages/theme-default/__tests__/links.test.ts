import { mdToSvelte } from '@sveltepress/vite'
import { describe, expect, it } from 'vitest'
import links from '../src/markdown/links'

const md = `
## Foo

[foo](https://www.google.com/)
[bar](/foo/bar)

`

describe('links', () => {
  it('simple', async () => {
    const r = await mdToSvelte({
      mdContent: md,
      filename: 'demo.md',
      remarkPlugins: [links, () => tree => {
        expect(JSON.stringify(tree, null, 2)).toMatchSnapshot()
      }],
    })
    expect(r).toMatchInlineSnapshot(`
      {
        "code": "<h2>Foo</h2>
      <p><Link to="https://www.google.com/" label="foo" />
      <Link to="/foo/bar" label="bar" /></p>",
        "data": {},
      }
    `)
  })
})
