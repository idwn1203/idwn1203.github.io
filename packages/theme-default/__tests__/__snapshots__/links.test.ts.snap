// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`links > simple 1`] = `
"{
  "type": "root",
  "children": [
    {
      "type": "heading",
      "depth": 2,
      "children": [
        {
          "type": "text",
          "value": "Foo",
          "position": {
            "start": {
              "line": 2,
              "column": 4,
              "offset": 4
            },
            "end": {
              "line": 2,
              "column": 7,
              "offset": 7
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 2,
          "column": 1,
          "offset": 1
        },
        "end": {
          "line": 2,
          "column": 7,
          "offset": 7
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "html",
          "value": "<Link to=\\"https://www.google.com/\\" label=\\"foo\\" />"
        },
        {
          "type": "text",
          "value": "\\n",
          "position": {
            "start": {
              "line": 4,
              "column": 31,
              "offset": 39
            },
            "end": {
              "line": 5,
              "column": 1,
              "offset": 40
            }
          }
        },
        {
          "type": "html",
          "value": "<Link to=\\"/foo/bar\\" label=\\"bar\\" />"
        }
      ],
      "position": {
        "start": {
          "line": 4,
          "column": 1,
          "offset": 9
        },
        "end": {
          "line": 5,
          "column": 16,
          "offset": 55
        }
      }
    }
  ],
  "position": {
    "start": {
      "line": 1,
      "column": 1,
      "offset": 0
    },
    "end": {
      "line": 7,
      "column": 1,
      "offset": 57
    }
  }
}"
`;
